File: Android/app/src/main/java/com/edu/primary/utils/InputValidator.java
package com.edu.primary.utils;

/**
 * 输入验证工具类（防御性编程）
 * 提供统一的输入验证方法
 */
public class InputValidator {
    
    private InputValidator() {
        // 工具类，禁止实例化
    }
    
    /**
     * 验证用户名
     * @param username 用户名
     * @return 是否有效
     */
    public static boolean isValidUsername(String username) {
        if (username == null || username.isEmpty()) {
            return false;
        }
        // 用户名长度3-20个字符，只能包含字母、数字、下划线
        return username.length() >= 3 && username.length() <= 20 
            && username.matches("^[a-zA-Z0-9_]+$");
    }
    
    /**
     * 验证密码
     * @param password 密码
     * @return 是否有效
     */
    public static boolean isValidPassword(String password) {
        if (password == null || password.isEmpty()) {
            return false;
        }
        // 密码长度至少6个字符
        return password.length() >= 6;
    }
    
    /**
     * 验证答案
     * @param answer 答案
     * @return 是否有效
     */
    public static boolean isValidAnswer(String answer) {
        return answer != null && !answer.trim().isEmpty();
    }
    
    /**
     * 验证年级
     * @param grade 年级
     * @return 是否有效
     */
    public static boolean isValidGrade(int grade) {
        return grade >= 1 && grade <= 6;
    }
    
    /**
     * 验证科目ID
     * @param subjectId 科目ID
     * @return 是否有效
     */
    public static boolean isValidSubjectId(int subjectId) {
        return subjectId >= 1 && subjectId <= 3;
    }
}

File: Android-1/app/src/main/java/com/edu/primary/util/InputValidator.kt
package com.edu.primary.util

/**
 * 输入校验：用户名（3–20 位字母数字下划线）、密码（至少 6 位）、年级（1–6）、学科 ID（1–3）。
 */
object InputValidator {
    /** 用户名非空、长度 3–20、仅字母数字下划线。 */
    fun isValidUsername(username: String?) = !username.isNullOrEmpty() &&
        username.length in 3..20 && username.matches(Regex("^[a-zA-Z0-9_]+$"))
    /** 密码非空且至少 6 位。 */
    fun isValidPassword(password: String?) = !password.isNullOrEmpty() && password.length >= 6
    /** 年级在 1–6 之间。 */
    fun isValidGrade(grade: Int) = grade in 1..6
    /** 学科 ID 在 1–3 之间（语数英）。 */
    fun isValidSubjectId(subjectId: Int) = subjectId in 1..3
}
